include/erlcloud_ecs.hrl

```diff
@@ -0,0 +1,306 @@
+    hard_limit :: pos_integer(),
+    name :: ecs_ulimit_name(),
+    soft_limit :: pos_integer()
+}).
+
+-record(ecs_volume_from, {
+    read_only :: boolean(),
+    source_container :: binary()
+}).
+
+-record(ecs_container_definition, {
+    command :: [binary()],
+    cpu :: pos_integer(),
+    disable_networking :: boolean(),
+    dns_search_domains :: [binary()],
+    docker_labels :: [{binary(), binary()}],
+    docker_security_options :: [binary()],
+    entry_point :: [binary()],
+    environment :: [#ecs_attribute{}],
+    essential :: boolean(),
+    extra_hosts :: [#ecs_host{}],
+    host_name :: binary(),
+    image :: binary(),
+    links :: [binary()],
+    log_configuration :: #ecs_log_configuration{},
+    memory :: pos_integer(),
+    memory_reservation :: pos_integer(),
+    mount_points :: [#ecs_mount_point{}],
+    name :: binary(),
+    port_mappings :: [#ecs_port_mapping{}],
+    privileged :: boolean(),
+    readonly_root_filesystem :: boolean(),
+    ulimits :: [#ecs_ulimit{}], 
+    user :: binary(),
+    volumes_from :: [#ecs_volume_from{}],
+    working_directory :: binary()
+}).
+
+-record(ecs_volume_host, {
+    source_path :: binary()
+}).
+
+-record(ecs_volume, {
+    host :: #ecs_volume_host{},
+    name :: binary()
+}).
+
+-record(ecs_task_definition, {
+    container_definitions :: [#ecs_container_definition{}],
+    family :: binary(),
+    network_mode :: binary(),
+    requires_attributes :: [#ecs_attribute{}],
+    revision :: pos_integer(),
+    status :: binary(),
+    task_definition_arn :: binary(),
+    task_role_arn :: binary(),
+    volumes :: [#ecs_volume{}]
+}).
+
+-record(ecs_network_binding, {
+    bind_ip :: binary(),
+    container_port :: pos_integer(),
+    host_port :: pos_integer(),
+    protocol :: ecs_protocol()
+}).
+
+-record(ecs_container, {
+    container_arn :: binary(),
+    exit_code :: pos_integer(),
+    last_status :: binary(),
+    name :: binary(),
+    network_bindings :: [#ecs_network_binding{}],
+    reason :: binary(),
+    task_arn :: binary()
+}).
+
+-record(ecs_container_override, {
+    command :: [binary()],
+    environment :: [#ecs_attribute{}],
+    name :: binary()
+}).
+
+-record(ecs_task_override, {
+    container_overrides :: [#ecs_container_override{}],
+    task_role_arn :: binary() 
+}).
+
+-record(ecs_task, {
+    cluster_arn :: binary(),
+    container_instance_arn :: binary(),
+    containers :: [#ecs_container{}],
+    created_at :: pos_integer(),
+    desired_status:: binary(),
+    last_status :: binary(),
+    overrides :: #ecs_task_override{},
+    started_at :: pos_integer(),
+    started_by :: binary(),
+    stopped_at :: pos_integer(),
+    stopped_reason :: binary(),
+    task_arn :: binary(),
+    task_definition_arn :: binary()
+}).
+
+-record(ecs_describe_tasks, {
+    tasks :: [#ecs_task{}],
+    failures :: [#ecs_failure{}]
+}).
+
+-record(ecs_list_clusters, {
+    cluster_arns :: [erlcloud_ecs:arn()],
+    next_token :: token()
```

Comment:
The point of separating token out is to have records for effective data and not having them for entire request/responses. I'm as a user of this library no longer interested in tokens inside records during further processing after list all is done.
Anyway if we'd like to keep records as returns let's rename `ecs_list_clusters` to `ecs_list_clusters_responce`, otherwise it sounds misleading to me because at some point we may also want to have records for input, for example `ecs_list_clusters_request`.